python/ray/data/_internal/execution/operators/actor_pool_submitter.py

```diff
@@ -0,0 +1,181 @@
+        """
+        # TODO(Clark): Better mapping from configured min/max pool size to static pool
+        # size?
+        pool_size = compute_strategy.max_size
+        if pool_size == float("inf"):
+            pool_size = compute_strategy.min_size
+        self._pool_size = pool_size
+        self._ray_remote_args = ray_remote_args
+        # A map from task output futures to the actors on which they are running.
+        self._active_actors: Dict[ObjectRef[Block], ray.actor.ActorHandle] = {}
+        # The actor pool on which we are running map tasks.
+        self._actor_pool = ActorPool()
+
+    def start(self):
+        # Check that actor pool hasn't yet been initialized.
+        assert self._actor_pool.num_actors == 0
+        # Create the actor workers and add them to the pool.
+        ray_remote_args = self._apply_default_remote_args(self._ray_remote_args)
+        cls_ = ray.remote(**ray_remote_args)(MapWorker)
+        for _ in range(self._pool_size):
+            self._actor_pool.add_actor(cls_.remote())
+
+    def submit(
+        self,
+        transform_fn: ObjectRef[Callable[[Iterator[Block]], Iterator[Block]]],
+        input_blocks: List[ObjectRef[Block]],
+    ) -> Tuple[ObjectRef[Block], ObjectRef[BlockMetadata]]:
+        # Pick an actor from the pool.
+        actor = self._actor_pool.pick_actor()
+        # Submit the map task.
+        block, block_metadata = actor.submit.options(num_returns=2).remote(
+            transform_fn, *input_blocks
+        )
+        self._active_actors[block] = actor
+        return block, block_metadata
+
+    def task_done(self, ref: ObjectRef[Block]):
+        # Return the actor that was running the task to the pool.
+        actor = self._active_actors.pop(ref)
+        self._actor_pool.return_actor(actor)
+
+    def task_submission_done(self):
+        # Kill all idle actors in the pool, and ensure that all remaining actors in the
+        # pool will be killed as they become idle.
+        self._actor_pool.kill_idle_actors()
+        self._actor_pool.kill_future_idle_actors()
+
+    def cancellable(self) -> bool:
+        # Actor tasks are not cancellable.
+        return False
+
+    @staticmethod
+    def _apply_default_remote_args(ray_remote_args: Dict[str, Any]) -> Dict[str, Any]:
+        """Apply defaults to the actor creation remote args."""
+        ray_remote_args = ray_remote_args.copy()
+        if "num_cpus" not in ray_remote_args:
+            ray_remote_args["num_cpus"] = 1
+        if "scheduling_strategy" not in ray_remote_args:
+            ctx = DatasetContext.get_current()
+            if ctx.scheduling_strategy == DEFAULT_SCHEDULING_STRATEGY:
+                ray_remote_args["scheduling_strategy"] = "SPREAD"
+            else:
+                ray_remote_args["scheduling_strategy"] = ctx.scheduling_strategy
+        return ray_remote_args
+
+
+class MapWorker:
+    """An actor worker for MapOperator."""
+
+    def ready(self):
+        return "ok"
```

Comment:
This is currently unused, since the actor pool is static and we don't trigger work submission or worker scale up on ready events.